### PR TITLE
Export boards functionality for backup #30

### PR DESCRIPTION
**Pull request by jquintozamora**
- Expose the 'boards' Redux state property in Containers/Board file
- Create a new React componet in Components/ExportButton
- Import said ExportButton to Containers/Board and add the component to the black toolbar

### CODE REVIEW

File: src/components/ExportButton/ExportButton.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        <FormattedMessage {...message} />
+    </Button>
+  );
+}
+
+ExportButton.propTypes = {
+  classes: PropTypes.object.isRequired,
+  className: PropTypes.string,
+  message: PropTypes.object,
+  handleExportClick: PropTypes.func,
```
Comment (shayc) [inline src/components/ExportButton/ExportButton.js:26]: by convention we use onSomethingEvent for props and handleSomethingEvent for event handler methods inside the component

File: src/containers/Board/Board.js [hunk 2]
```diff
@@ -165,6 +166,24 @@ export class Board extends Component {
...
 
+  handleExportClick = () => {
+    const exportFilename = "board.json";
+    const { boards } = this.props;
+    const jsonData = new Blob([JSON.stringify(boards)], {type: 'text/json;charset=utf-8;'});
+    //IE11 & Edge
+    if (navigator.msSaveBlob) {
+        navigator.msSaveBlob(jsonData, exportFilename);
+    } else {
+        //In FF link must be added to DOM to be clicked
```
Comment (shayc) [inline src/containers/Board/Board.js:177]: wow, I didn't know that, way to go mate for implementing!

File: src/containers/Board/Board.js [hunk 3]
```diff
@@ -263,7 +282,7 @@ export class Board extends Component {
           </div>
           <div className="Toolbar__group Toolbar__group--end">
             {this.state.isSelecting && <div />}
-
+            <ExportButton message={messages.export} handleExportClick={this.handleExportClick} />
```
Comment (shayc) [inline src/containers/Board/Board.js:285]: handleExportClick should be changed to onExportClick

**Review by shayc — COMMENTED** [review]
good job! only thing is the naming convention and we can merge, I'd be happy to do it for you, let me know!

File: src/containers/Board/messages.js [hunk 4]
```diff
@@ -28,5 +28,9 @@ export default defineMessages({
   add: {
     id: 'cboard.containers.App.add',
     defaultMessage: 'Add'
+  },
+  export: {
+    id: 'cboard.containers.App.export',
```
Comment (shayc) [inline src/containers/Board/messages.js:33]: I'll add that string to our crowdin.com project

**Comment by shayc** [comment]
ok merged 👍

**Comment by shayc** [comment]
thanks!

File: src/components/ExportButton/ExportButton.js (hunk 1 above)
Comment (jquintozamora) [reply src/components/ExportButton/ExportButton.js:26]: I didn´t know! thanks for the comment!

**Comment by shayc** [comment]
After merging your pull request I noticed the location of the ExportButton on the toolbar was a bad choice (by me), it messed up the UI for mobile phone users (not enough room). I had to make a new Settings/Backup screen and place it there. It is deployed, and you can have a look.

**Comment by jquintozamora** [comment]
That´s perfect, and it answers my next question, which is: where should I put the button for importing? :)

**Comment by shayc** [comment]
Import component should be referenced in Backup.js, Backup.js should be connected to Redux so it can restore the boards property on the Board state, you'll need to create new action and new action creator. Let me know if you need help with anything :)

**Pull request merged**